src/lua/LuaInput.cpp

```diff
@@ -321,6 +362,178 @@ static int l_input_set_joystick_enabled(lua_State *l)
 	return 0;
 }
 
+/*
+ * Function: ActionOnPress
+ *
+ * Activate the action corresponding to the given 'LuaInputAction' object
+ *
+ * Example:
+ *
+ * > action = Input.GetActionBinding("BindToggleSpeedLimiter")
+ * > if ui.isMouseClicked(0) then
+ * >     Input.ActionOnPress(action)
+ * > elseif ui.isMouseReleased(0) then
+ * >     Input.ActionOnRelease(action)
+ * > end
+ *
+ * Parameters:
+ *
+ *   action: a LuaInputAction object, obtained via Input.GetActionBinding
+ *
+ * Returns:
+ *
+ *   nothing
+ *
+ */
+static int l_input_action_on_press(lua_State *l)
+{
+	auto action = LuaObject<LuaInputAction>::CheckFromLua(1)->getAction();
+	action->binding.m_queuedEvents |= 1;
+	action->binding.m_active = true;
+	return 0;
+}
```

Comment:
I think this (and most of the remaining action/axis manipulation functions added in this PR) should be a member function on the action object itself; e.g. `binding.action:OnPress()` or `binding.action:IsActive()`. While I do like how easy it is to search for `Input.IsActionActive`, it's unwieldy to type and logically the effect of these functions is limited to the action/axis object itself so it is a good fit for a member function.

Reply:
Done! It's nice that this change allowed to remove the `require 'Input'` from several modules.